### PR TITLE
Adding unit test for the mcmc sampling

### PR DESCRIPTION
**Pull request by facero**
**Description**

This pull request adds unit tests for the sampling MCMC.
With few runs for the MCMC, the full tests take only 3sec

Minor question:
I tried to use pytest fixture to avoid having to load the datasets in each test but this didn't work as I expected. Can I use fixture here to load the dataset for all functions here ?

### CODE REVIEW

**Comment by facero** [comment]
The tests passed on my machine but apparently not in here due to `ModuleNotFoundError: No module named 'emcee'`.
Is it possible to add it to the pipeline configuration ?

**Comment by registerrier** [comment]
I think you need to specify that the dependency is required:
`from gammapy.utils.testing import requires_dependency`

and before the function:
`@requires_dependency("emcee")`

File: gammapy/modeling/tests/test_sampling.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from gammapy.datasets import Datasets
+from gammapy.modeling.sampling import run_mcmc, ln_uniform_prior
+import numpy as np
+from gammapy.utils.testing import requires_dependency
```
Comment (registerrier) [inline gammapy/modeling/tests/test_sampling.py:5]: Here you also need to import `requires_data`

File: gammapy/modeling/tests/test_sampling.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    parameters["lat_0"].min = -90
+    parameters["lat_0"].max = 90
+    parameters["lon_0"].min = 0
+    parameters["lon_0"].max = 360
+    parameters["amplitude"].min = 0.01 * parameters["amplitude"].value
+    parameters["amplitude"].max = 100 * parameters["amplitude"].value
+
+    return dataset
+
+def test_lnprob():
```
Comment (registerrier) [inline gammapy/modeling/tests/test_sampling.py:27]: Here add the decorator: `@requires_data()`

File: gammapy/modeling/tests/test_sampling.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    parameters = dataset.models.parameters
+
+    #paramater is within min, max boundaries
+    assert ln_uniform_prior(dataset) == 0.0
+    # Setting amplitude outside min, max values
+    parameters["amplitude"].value = 1000
+    assert ln_uniform_prior(dataset) == -np.inf
+
+@requires_dependency("emcee")
+def test_runmcmc():
```
Comment (registerrier) [inline gammapy/modeling/tests/test_sampling.py:39]: Here you also need the decorator: `@requires_data()`

File: gammapy/modeling/tests/test_sampling.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    #Running a small MCMC on pregenerated datasets
+    dataset = get_datasets()
+    run_mcmc(dataset, nwalkers=6, nrun=10)  # to speedup the test
+
```
Comment (registerrier) [inline gammapy/modeling/tests/test_sampling.py:43]: There is nothing to assert here?

File: gammapy/modeling/tests/test_sampling.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from gammapy.datasets import Datasets
+from gammapy.modeling.sampling import run_mcmc, ln_uniform_prior
+import numpy as np
```
Comment (registerrier) [inline gammapy/modeling/tests/test_sampling.py:4]: Import non gammapy dependencies first

File: gammapy/modeling/tests/test_sampling.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+from gammapy.utils.testing import requires_dependency
+
+def get_datasets():
```
Comment (registerrier) [inline gammapy/modeling/tests/test_sampling.py:7]: You might want to use a purest fixture here.
First you need to `import pytest`

then rewrite the definition of the function:
```
@pytest.fixture(scope="session")
def dataset():
```

then add as a argument for the test functions : `dataset`
i.e. `def test_lnprob(dataset)`
and 
`def test_runmcmc(dataset) `

File: gammapy/modeling/tests/test_sampling.py (hunk 4 above)
Comment (facero) [reply gammapy/modeling/tests/test_sampling.py:49]: I added assert that the sampler returned is an emcee.ensemble.EnsembleSampler  object.
As the walker are random hard to test any specific value.

File: gammapy/modeling/tests/test_sampling.py (hunk 6 above)
Comment (facero) [reply gammapy/modeling/tests/test_sampling.py]: thanks, now I was able to use pytest fixtures !

**Pull request merged**